helpdesk/views/staff.py

```diff
@@ -422,8 +424,16 @@ def view_ticket(request, ticket_id):
 
         return redirect('helpdesk:edit_ticket_checklist', ticket.id, checklist.id)
 
+    open_dependencies = ticket.ticketdependency.filter(depends_on__status__in=Ticket.OPEN_STATUSES)
+    dependencies = [d for d in itertools.chain(
+        open_dependencies,
+        ticket.ticketdependency.all().difference(open_dependencies)
+    )]
```

Comment:
I don't understand why you need to do this. Isn't it possible to simply do :

```suggestion
    dependencies = ticket.ticketdependency.all()
```

Maybe you wish to keep open statuses at the beginning?

Reply:
Yes: It's about ordering. I use helpdesk (also) for tracking software development/maintenance. With the ticket dependencies I can model milestones, but I want to trickle resolved/closed tickets down the list, so I can pick always from top.

![image](https://github.com/django-helpdesk/django-helpdesk/assets/20359147/c7fb09cc-3a99-4188-acb8-c2bebb3c4fd0)

I thought about ordering by status first, just like you point it out. But  since ticket status can be configured / named freely ordering might not
in all cases put open tickets on top. In contrast, using the TICKET_OPEN_STATUSES settings is a stable approach.

The itertools trick allowed me to keep the template code tidy, since there is only one loop to go through - or an empty list.  The implementation is based on the assumption, that the number of dependent ticktes is not huge.  I would have preferred a QueryList, but could not came up with an order preserving expression.

Another side topic would be to sort the two lists (open/closed tickets) individually before creating the list, but then I did not have a specific preference and left it out.

Reply:
Given that status values can be overridden and there is no directive to specify status values to specifically accommodate ordering so that the open statuses appear first, it may be better to enhance the operators available in the ORM to include "NOT IN" to retrieve the unopened dependencies like this:
`ticket.ticketdependency.filter(depends_on__status__not_in=Ticket.OPEN_STATUSES)`
Then you simplify the code to this:
```
    open_dependencies = ticket.ticketdependency.filter(depends_on__status__in=Ticket.OPEN_STATUSES)
    dependencies = open_dependencies + ticket.ticketdependency.filter(depends_on__status__not_in=Ticket.OPEN_STATUSES)
```

To do this we add this class to a module helpdesk/orm_operators.py:
```
from django.db.models.lookups import In

class NotIn(In):
    lookup_name = "not_in"

    def get_rhs_op(self, connection, rhs):
        return "NOT IN %s" % rhs
```

Then make the new operator available in the ORM by adding this to the ready method in helpdesk/apps.py:
```
        from django.db.models.fields import Field
        from helpdesk.orm_operators import NotIn
        Field.register_lookup(NotEqual)
```

Reply:
Excelent, will do that.

Reply:
The `+` does not work on querysets, but it seems to work with `|`

Reply:
`|` is not order preserving. The current implementation still uses NotIn, but concatenates via list()+list()